src/core/qgseditformconfig.cpp

```diff
@@ -660,20 +661,36 @@ QgsAttributeEditorElement *QgsEditFormConfig::attributeEditorElementFromDomEleme
     // At this time, the relations are not loaded
     // So we only grab the id and delegate the rest to onRelationsLoaded()
     QgsAttributeEditorRelation *relElement = new QgsAttributeEditorRelation( elem.attribute( QStringLiteral( "relation" ), QStringLiteral( "[None]" ) ), parent );
-    if ( elem.hasAttribute( "buttons" ) )
-    {
-      QString buttonString = elem.attribute( QStringLiteral( "buttons" ), qgsFlagValueToKeys( QgsAttributeEditorRelation::Button::AllButtons ) );
-      relElement->setVisibleButtons( qgsFlagKeysToValue( buttonString, QgsAttributeEditorRelation::Button::AllButtons ) );
-    }
-    else
+    QVariantMap config = QgsXmlUtils::readVariant( elem.firstChildElement( "config" ) ).toMap();
+
+    // load defaults
+    if ( config.isEmpty() )
+      config = relElement->config();
+
+    // pre QGIS 3.18 compatibility
+    Q_NOWARN_DEPRECATED_PUSH
```

Comment:
Can you rework this to avoid the deprecated function calls? We should only use Q_NOWARN_DEPRECATED_PUSH for code left in place for api compatibility, not for project compatibility. Otherwise when 4.0 rolls around and someone removes the deprecated calls they'll either have to do that work themselves, or drop the old project compatibility (not nice either way)

Reply:
Thanks, done